### ISSUE TITLE
Gas consumption ? 

### ISSUE THREAD

**Issue by edelans** [Aohzan/ecodevices#12]
Hello @Aohzan !

# Thanks 🙏 

First of all, I want to say thanks for your plugin, I'm super happy to have found it, it works like a charm. 

# Request

I have a "[compteur gazpard](https://www.grdf.fr/particuliers/fonctionnement-compteur-gaz-communicant-grdf)" in my house, that I successfully integrated with my ecodevice : 

![CleanShot 2021-09-08 at 09 15 06](https://user-images.githubusercontent.com/2805640/132463671-d8455d36-1c37-4543-aa54-d6eaa48406a7.png)

and into HA thanks to your plugin : 

 
![CleanShot 2021-09-08 at 09 10 49@2x](https://user-images.githubusercontent.com/2805640/132463037-2b4fc6eb-1c0a-45d7-b73e-18cb014b90fa.png)

But I cannot use this entity to configure the gas consumption in the energy settings : 

<img width="463" alt="CleanShot 2021-09-08 at 09 06 39@2x" src="https://user-images.githubusercontent.com/2805640/132462796-1fe974bf-e495-4962-937f-68243184bb43.png">

While I could configure the data from the linky (teleinfo) in the electricity tracking panel of the energy dashboard.

By any chance, do you know how I can make this work ?

Thanks again ! 

--
btw, I don't remember where I saw that (old gh profile maybe ?), but I think you live in Rennes (me too !) : if it's still the case, I'll be glad to offer you a beer 🍻

**Comment by Aohzan** [comment]
Hi !
I think you must set `m³` as unit of measurement for the Meter 2 Total. It's not possible to set a different unit between the daily and total  sensor currently for meter entities.
Temporary you can do it with https://www.home-assistant.io/docs/configuration/customizing-devices/#manual-customization
I will add the feature shortly

(ok for the beer :grin:)

**Comment by edelans** [comment]
ok, FWIW, I made it work by creating a new sensor by adding the following lines in my configuration.yaml file : 

```
template:
  - sensor:
      name: Gas Consumption m3
      state: >-
        {{ states('sensor.gaz')|float / 1000 }}
      icon: mdi:fire
      unit_of_measurement: "m³"
      state_class: "total_increasing"
      device_class: "gas"
```

I shoot you an email for the beer ;)

**Comment by edelans** [comment]
Improvement : as I encountered the same problem as in https://github.com/Aohzan/ecodevices/issues/13 with some zero values wreaking havoc in the gas consumption dashboard, I also embedded a filter to only keep increasing values in the new sensor : 

```
  - sensor:
      name: Gas Consumption m3 increasing only
      state: >
        {% if states('sensor.gaz') > states('sensor.gas_consumption_m3_increasing_only') %}
          {{ states('sensor.gaz') | float(states('sensor.gas_consumption_m3_increasing_only' * 1000)) / 1000 }}
        {% else %} 
          {{ states('sensor.gas_consumption_m3_increasing_only') }}
        {% endif %}
      icon: mdi:fire
      unit_of_measurement: "m³"
      state_class: "total_increasing"
      device_class: "gas"
```

**Comment by Aohzan** [comment]
Hi,
You can now define a different unit of measurement for the total sensor : https://github.com/Aohzan/ecodevices/tree/2.5.0b

**Comment by Aohzan** [comment]
fix in last 3.0.0 version
